### ISSUE TITLE
.Error dialing TCP:  **  connectex: A connection attempt failed because the connected party did not properly respond

### ISSUE THREAD

**Issue by pivan9911** [minishift/minishift#2440]
### General information

  * Minishift version:  v1.18.0+756b128

  * OS:  Windows 10

  * Hypervisor:  Hyper-V /

### Steps to reproduce

  1.   minishift stop
  2.   minishift delete --force
  3.   minishift --memory 2G --show-libmachine-logs -v5 start
  

### Expected
I was  hoping it would start,  and  minishich- command would work
  e.g.     minishift ssh -- docket ps 

### Actual
I'm getting problems with TCP,  please see also the log.
 "Connection attempt failed because the connected party did not properly "

My McAfee - firewall is turned off, 
The Hyper-V is set on "Private-network".

### Logs
```
PS C:\Users\openshift> minishift stop
Stopping local OpenShift cluster...
Cluster stopped.

PS C:\Users\openshift> minishift delete --force
Deleting the Minishift VM...
Minishift VM deleted.

PS C:\Users\openshift> minishift --memory 2G --show-libmachine-logs -v5 start
-- minishift version: v1.18.0+756b128
-- Starting profile 'minishift'
Found binary path at C:\Users\openshift\Hyper-V\minishift-1.18.0\minishift.exe
Launching plugin server for driver hyperv
Plugin server listening at address 127.0.0.1:53573
() Calling .GetVersion
Using API Version  1
() Calling .SetConfigRaw
() Calling .GetMachineName
(minishift) Calling .GetState
(minishift) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive ( Get-VM minishift ).state
(minishift) DBG | [stdout =====>] : Off
(minishift) DBG |
-- Checking if https://github.com is reachable ... (minishift) DBG | [stderr =====>] :
OK
-- Checking if requested OpenShift version 'v3.9.0' is valid ... OK
-- Checking if requested OpenShift version 'v3.9.0' is supported ... OK
-- Checking if requested hypervisor 'hyperv' is supported on this platform ... OK
-- Checking if Hyper-V driver is installed ... OK
-- Checking if Hyper-V driver is configured to use a Virtual Switch ...
   'minishift-switch' ... OK
-- Checking if user is a member of the Hyper-V Administrators group ... OK
-- Checking the ISO URL ... OK
-- Checking if provided oc flags are supported ... OK
-- Starting local OpenShift cluster using 'hyperv' hypervisor ...
-- Starting Minishift VM ....Found binary path at C:\Users\openshift\Hyper-V\minishift-1.18.0\minishift.exe
Launching plugin server for driver hyperv
Plugin server listening at address 127.0.0.1:53579
() Calling .GetVersion
Using API Version  1
() Calling .SetConfigRaw
() Calling .GetMachineName
(minishift) Calling .GetState
(minishift) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive ( Get-VM minishift ).state
(minishift) DBG | [stdout =====>] : Off
(minishift) DBG |
(minishift) Calling .Start
(minishift) DBG | [stderr =====>] :
(minishift) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive Start-VM minishift
.....(minishift) DBG | [stdout =====>] :
(minishift) Waiting for host to start...
(minishift) DBG | [stderr =====>] :
(minishift) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive ( Get-VM minishift ).state
(minishift) DBG | [stdout =====>] : Running
(minishift) DBG |
(minishift) DBG | [stderr =====>] :
(minishift) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive (( Get-VM minishift ).networkadapters[0]).ipaddresses[0]
.(minishift) DBG | [stdout =====>] :
(minishift) DBG | [stderr =====>] :
(minishift) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive ( Get-VM minishift ).state
.(minishift) DBG | [stdout =====>] : Running
(minishift) DBG |
(minishift) DBG | [stderr =====>] :
(minishift) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive (( Get-VM minishift ).networkadapters[0]).ipaddresses[0]
(minishift) DBG | [stdout =====>] :
(minishift) DBG | [stderr =====>] :
.(minishift) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive ( Get-VM minishift ).state
(minishift) DBG | [stdout =====>] : Running
(minishift) DBG |
(minishift) DBG | [stderr =====>] :
(minishift) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive (( Get-VM minishift ).networkadapters[0]).ipaddresses[0]
(minishift) DBG | [stdout =====>] :
(minishift) DBG | [stderr =====>] :
.(minishift) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive ( Get-VM minishift ).state
(minishift) DBG | [stdout =====>] : Running
(minishift) DBG |
(minishift) DBG | [stderr =====>] :
(minishift) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive (( Get-VM minishift ).networkadapters[0]).ipaddresses[0]
.(minishift) DBG | [stdout =====>] :
(minishift) DBG | [stderr =====>] :
(minishift) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive ( Get-VM minishift ).state
(minishift) DBG | [stdout =====>] : Running
(minishift) DBG |
(minishift) DBG | [stderr =====>] :
(minishift) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive (( Get-VM minishift ).networkadapters[0]).ipaddresses[0]
(minishift) DBG | [stdout =====>] :
(minishift) DBG | [stderr =====>] :
.(minishift) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive ( Get-VM minishift ).state
(minishift) DBG | [stdout =====>] : Running
(minishift) DBG |
(minishift) DBG | [stderr =====>] :
(minishift) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive (( Get-VM minishift ).networkadapters[0]).ipaddresses[0]
(minishift) DBG | [stdout =====>] :
(minishift) DBG | [stderr =====>] :
.(minishift) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive ( Get-VM minishift ).state
(minishift) DBG | [stdout =====>] : Running
(minishift) DBG |
(minishift) DBG | [stderr =====>] :
(minishift) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive (( Get-VM minishift ).networkadapters[0]).ipaddresses[0]
(minishift) DBG | [stdout =====>] :
(minishift) DBG | [stderr =====>] :
.(minishift) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive ( Get-VM minishift ).state
(minishift) DBG | [stdout =====>] : Running
(minishift) DBG |
(minishift) DBG | [stderr =====>] :
(minishift) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive (( Get-VM minishift ).networkadapters[0]).ipaddresses[0]
(minishift) DBG | [stdout =====>] :
(minishift) DBG | [stderr =====>] :
.(minishift) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive ( Get-VM minishift ).state
(minishift) DBG | [stdout =====>] : Running
(minishift) DBG |
(minishift) DBG | [stderr =====>] :
(minishift) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive (( Get-VM minishift ).networkadapters[0]).ipaddresses[0]
(minishift) DBG | [stdout =====>] : fe80::215:5dff:fe00:7a0a
(minishift) DBG |
(minishift) DBG | [stderr =====>] :
(minishift) Calling .GetConfigRaw
Waiting for SSH to be available...
Getting to WaitForSSH function...
(minishift) Calling .GetSSHHostname
(minishift) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive ( Get-VM minishift ).state
(minishift) DBG | [stdout =====>] : Running
(minishift) DBG |
(minishift) DBG | [stderr =====>] :
(minishift) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive (( Get-VM minishift ).networkadapters[0]).ipaddresses[0]
.(minishift) DBG | [stdout =====>] : fe80::215:5dff:fe00:7a0a
(minishift) DBG |
(minishift) DBG | [stderr =====>] :
(minishift) Calling .GetSSHPort
(minishift) Calling .GetSSHKeyPath
(minishift) Calling .GetSSHKeyPath
(minishift) Calling .GetSSHUsername
SSH binary not found, using native Go implementation
&{{{<nil> 0 [] [] []} docker [0x7b5120] 0x7b50d0 <nil>  [] 0s} fe80::215:5dff:fe00:7a0a 22 <nil> <nil>}
About to run SSH command:
exit 0
......Error dialing TCP: dial tcp [fe80::215:5dff:fe00:7a0a]:22: connectex: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
......Error dialing TCP: dial tcp [fe80::215:5dff:fe00:7a0a]:22: connectex: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
```

**Comment by pivan9911** [comment]
I have set the Virtual Proxy to External, as in Picture.
![hyperv_external](https://user-images.githubusercontent.com/13240295/41039712-9aa0fc8e-699a-11e8-8115-f12038de3695.JPG)

And restarted minishift, with better luck.

Now I'm getting:
\Users\openshift> minishift ssh -- docker ps
Command failed: dial tcp 172.17.4.98:22: connectex: No connection could be made because the target machine actively refused it.

Most likely Firewall-issue.  The network 172.17.4.98 is visible in my routes,  
[RouteExternal.txt](https://github.com/minishift/minishift/files/2076368/RouteExternal.txt)

How does one, open McAfee firewalls for ssh and all other tools(Browsers https://172.17.4.98:8443/) 
that need access to this network: 172.17.4.98   ??

Kind Regards
  IvnP

**Comment by gbraad** [comment]
It is very likely a networking/firewall issue. I do not know much about McAfee, so can't help you there...

**Comment by gbraad** [comment]
Note: The Hyper-V is set on "Private-network". will ONLY work if you also provide a local DHCP server yourself, or start minishift with the Static IP assignment, `--network-ipaddress`, `--network-nameserver`, etc

**Comment by pivan9911** [comment]
I changed Hyper-V to "Default Switch",  which is "Internal Network", and now the "minishift start" 
failes on:  
------------------------------------------
Applying addon 'admin-user':..Error during post cluster up configuration: 
Error executing command 'oc adm policy add-cluster-role-to-user cluster-admin admin'.
-------------------------------------------

Command like   "oc get pods"    give error:
--------------------------------
Error from server (Forbidden): pods "9f67ed7c2cf0" is forbidden:
         User "dev" cannot get pods in the namespace "default": 
         User "dev" cannot get pods in project "default"
-------------------------------------

Is it better with    "Private-network"   plus   "--network-ipaddress, --network-nameserver" ??
How do I find which  --network-ipaddress   should be specified ?
 
It's a pitty, not to have a Windows-10 log,    from a succesfull     
  >>  minishift --vm-driver hyperv  --show-libmachine-logs -v8 start  

would be nice to see, it actually can succeed  :-)

Thanks a lot, for the feedback
  pivan

**Comment by gbraad** [comment]
Deployments on HyperV can succeed on either network, as long as correctly
configured. I am mostly working on this and redeploy several times daily as
part of the development.

But... Be sure to use `minishift delete` between changes. I suggest to use
an external virtual switch first. This however needs the network you
connect to to offer DHCP.

During start, please see the progress using `minishift
--show-libmachine-logs -v5`.


-- 

   Gerard Braad | http://gbraad.nl
   [ Doing Open Source Matters ]

**Comment by pivan9911** [comment]
[Minishift_Start4_private.txt](https://github.com/minishift/minishift/files/2098867/Minishift_Start4_private.txt)

**Comment by pivan9911** [comment]
In previous comment is shown log of   "minishift start"

Is it succesfull??     I don't know,?  I would need to see one, to  keep my motivation :-)
 
In the log.  
  failing   "oc get pods"        
   User "system" cannot list pods in the
   Which  user   is allowed to do this ??

 failing   "pidof dockerd-current"  
is there no docker-daemon ??

Many questions.

Kind regards
  I.P.

**Comment by gbraad** [comment]
Hard to follow with all the debug, but likely.


OpenShift server started.

The server is accessible via web console at:
   https://172.31.216.45:8443




-- 

   Gerard Braad | http://gbraad.nl
   [ Doing Open Source Matters ]

**Comment by budhrg** [comment]
Any update here @pivan9911 ?

**Comment by jessiegocotano** [comment]
something wrong here: `[fe80::215:5dff:fe00:7a0a]:22`
maybe typo or ip changed?

**Comment by gbraad** [comment]
That is an IPv6 address. Likely you have issues with your network
configuration.

**Comment by craigwhiteman-enwl** [comment]
I had this and a multitude of other issues with my minishift. I am using IPv4, the Default Switch in Hyper V on Windows 10 and I my network connection is WiFi.  It continuously failed to work even with **minishift delete --force --clear-cache**. It finally started working and the only difference was that the failures were using the IP address 192.168.x.x whereas the successful one was using 172.x.x.x. My WiFi switch uses the 192.168.x.x network so it seems there was a clash between this and minishift.
